src/entities/Balance/utils.ts

```diff
@@ -0,0 +1,161 @@
+import { ChainId } from '@dcl/schemas'
+import { TokenBalancesResponse, Alchemy, TokenBalanceResponse } from '../../api/Alchemy'
+import WalletModel from '../Wallet/model'
+import { WalletAttributes } from '../Wallet/types'
+import { TokenAttributes } from '../Token/types'
+import TokenModel from '../Token/model'
+import { asNumber } from '../Proposal/utils'
+import { BalanceAttributes, AggregatedTokenBalance, TokenBalance } from './types'
+import BalanceModel from './model'
+import { v1 as uuid } from 'uuid'
+import logger from 'decentraland-gatsby/dist/entities/Development/logger'
+import Time from 'decentraland-gatsby/dist/utils/date/Time'
+import { formattedTokenBalance } from '../Wallet/utils'
+
+export const NATIVE_CONTRACT = 'NATIVE'
+
+export async function getNativeBalance(wallet: WalletAttributes) {
+  const chainId = wallet.network
+  const nativeBalance = await Alchemy.get(chainId).getNativeBalances(wallet.address)
+  let nativeBalanceResult: TokenBalanceResponse = {
+    'contractAddress': NATIVE_CONTRACT,
+    'tokenBalance': nativeBalance.result,
+    'error': null
+  }
+  return nativeBalanceResult
+}
+
+export async function getTokenBalancesFor(wallet: WalletAttributes): Promise<TokenBalanceResponse[]> {
+  const chainId = wallet.network
+  const contracts: string[] = await TokenModel.getContracts(chainId)
+  const balances: TokenBalancesResponse = await Alchemy.get(chainId).getTokenBalances(wallet.address, contracts)
+  const tokenBalances = balances.result.tokenBalances
+
+  const nativeBalanceResult = await getNativeBalance(wallet)
+  tokenBalances.push(nativeBalanceResult)
+
+  return tokenBalances.filter(balance => {
+    if (balance.error === null) {
+      return true
+    } else {
+      logger.error('Error while fetching token balance', {
+        wallet: wallet,
+        token: balance.contractAddress,
+        error: balance.error
+      })
+      return false
+    }
+  });
+}
+
+export async function getBlockNumber(chainId: ChainId) {
+  const blockNumber = await Alchemy.get(chainId).getBlockNumber()
+  return blockNumber.result
+}
+
+export async function latestConsistentBlockNumber(chainId: ChainId) {
+  let currentBlockNumber = await getBlockNumber(chainId)
+  return (asNumber(currentBlockNumber) - asNumber(0x100)).toString(16)
+}
+
+export async function getBalances() {
+  const createdBalances = []
+  const errors = []
+
+  const wallets: WalletAttributes[] = await WalletModel.find()
+
+  for (const wallet of wallets) {
+    const tokenBalances: TokenBalanceResponse[] = await getTokenBalancesFor(wallet)
+    for (const tokenBalance of tokenBalances) {
+      try {
+        const token = await TokenModel.findMatchingToken(wallet.network, tokenBalance.contractAddress)
+        const previousBalance = await BalanceModel.findLatest(wallet.id, token.id)
+        if (previousBalance && previousBalance.amount === tokenBalance.tokenBalance) {
+          return
```

Comment:
would't this ends the execution of the function and return `undefined` if found only one balance that match the previous one? seems like an unexpected side-effecs